libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -63,68 +61,29 @@ void UARTDriver::begin(uint32_t b)
 
 void UARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
 {
-    if (device_path == NULL && _console) {
+    if (device_path == NULL && _console && !_initialised) {
         _device = new ConsoleDevice();
-        _device->open();
-        _device->set_blocking(false);
     } else if (!_initialised) {
         if (device_path == NULL) {
             return;
         }
 
-        switch (_parseDevicePath(device_path)) {
-        case DEVICE_TCP:
-        {
-            _tcp_start_connection();
-            _flow_control = FLOW_CONTROL_ENABLE;
-            break;
-        }
-
-        case DEVICE_UDP:
-        {
-            _udp_start_connection();
-            _flow_control = FLOW_CONTROL_ENABLE;
-            break;
-        }
-
-        case DEVICE_UDPIN:
-        {
-            _udpin_start_connection();
-            _flow_control = FLOW_CONTROL_ENABLE;
-            break;
-        }
-        
-#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_QFLIGHT
-        case DEVICE_QFLIGHT:
-        {
-            _qflight_start_connection();
-            _flow_control = FLOW_CONTROL_DISABLE;
-            break;
-        }
-#endif
+        _device = _parseDevicePath(device_path);
```

Comment:
He changed _device to be OwnPtr. This means that when you assign to something else it internally destroys the previous object. See the unit test for OwnPtr, it's like std::unique_ptr